mutalyzer/backtranslator.py

```diff
@@ -0,0 +1,139 @@
+"""
+Mutalyzer interface for variant back translation (p. to c.).
+"""
+
+
+from __future__ import unicode_literals
+
+from backtranslate.backtranslate import BackTranslate
+from backtranslate.util import protein_letters_3to1
+from Bio.Data import CodonTable
+from extractor.variant import Allele, DNAVar, ISeq, ISeqList
+
+from . import ncbi, Retriever
+from .grammar import Grammar
+from .output import Output
+
+
+def backtranslate(output, description, table_id=1):
+    """
+    Back translation of an amino acid substitution to all possible
+    causal one-nucleotide substitutions.
+
+    :arg object output: Output object for logging.
+    :arg unicode description: Amino acid substitution in HGVS format.
+    :arg int table_id: Translation table id.
+
+    :returns: List of DNA substitutions in HGVS format.
+    :rtype: list(unicode)
+    """
+    # TODO:
+    # - Support reference sequences other than transcript or protein.
+    # - Check input:
+    #   - without_dna: Does the reference amino acid fit?
+
+    #
+    # Observations regarding the HGVS parser:
+    # - p.AspxxTyr is printed differently than c.yyA>G
+    # - Allele descriptions are not supported?
```

Comment:
Also non-blocking, but these were some notes I made.
1) Both substitutions (c. and p.) have an `__str__` representation. I just noticed that they are not aligned.
2) Apparently, a variant like `p.[Met1Arg;Glu10Ter]` is not supported.